packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-glue-job-task.ts

```diff
@@ -0,0 +1,108 @@
+   *
+   * The valid value for Glue is either FIRE_AND_FORGET or SYNC.
+   *
+   * @default FIRE_AND_FORGET
+   */
+  readonly integrationPattern?: sfn.ServiceIntegrationPattern;
+
+  /**
+   * The job arguments specifically for this run.
+   *
+   * For this job run, they replace the default arguments set in the job definition itself.
+   *
+   * @default - Default arguments set in the job definition
+   */
+  readonly arguments?: { [key: string]: string };
+
+  /**
+   * The job run timeout.
+   *
+   * This is the maximum time that a job run can consume resources before it is terminated and enters TIMEOUT status.
+   * Must be at least 1 minute.
+   *
+   * @default - Default timeout set in the job definition
+   */
+  readonly timeout?: Duration;
+
+  /**
+   * The name of the SecurityConfiguration structure to be used with this job run.
+   *
+   * This must match the Glue API
+   * [single-line string pattern](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-common.html#aws-glue-api-regex-oneLine).
+   *
+   * @default - Default configuration set in the job definition
+   */
+  readonly securityConfiguration?: string;
+
+  /**
+   * After a job run starts, the number of minutes to wait before sending a job run delay notification.
+   *
+   * Must be at least 1 minute.
+   *
+   * @default - Default delay set in the job definition
+   */
+  readonly notifyDelayAfter?: Duration;
+}
+
+/**
+ * Invoke a Glue job as a Task
+ *
+ * OUTPUT: the output of this task is a JobRun structure, for details consult
+ * https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-jobs-runs.html#aws-glue-api-jobs-runs-JobRun
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-glue.html
+ */
+export class RunGlueJobTask implements sfn.IStepFunctionsTask {
+  private readonly integrationPattern: sfn.ServiceIntegrationPattern;
+
+  constructor(private readonly glueJobName: string, private readonly props: RunGlueJobTaskProps = {}) {
+    this.integrationPattern = props.integrationPattern || sfn.ServiceIntegrationPattern.FIRE_AND_FORGET;
+
+    const supportedPatterns = [
+      sfn.ServiceIntegrationPattern.FIRE_AND_FORGET,
+      sfn.ServiceIntegrationPattern.SYNC
+    ];
+
+    if (!supportedPatterns.includes(this.integrationPattern)) {
+      throw new Error(`Invalid Service Integration Pattern: ${this.integrationPattern} is not supported to call Glue.`);
+    }
+  }
+
+  public bind(_task: sfn.Task): sfn.StepFunctionsTaskConfig {
+    const notificationProperty = this.props.notifyDelayAfter ? { NotifyDelayAfter: this.props.notifyDelayAfter.toMinutes() } : null;
+    return {
+      resourceArn: getResourceArn("glue", "startJobRun", this.integrationPattern),
+      policyStatements: [new iam.PolicyStatement({
+        resources: ["*"],
+        actions: [
+          "glue:StartJobRun",
+          "glue:GetJobRun",
+          "glue:GetJobRuns",
+          "glue:BatchStopJobRun"
+        ],
```

Comment:
Sorry, there's something I missed here. 

Can we reduce the scope of this to just the glue job's ARN? 

Also, why does this task need `BatchStopJobRun` access?

Reply:
I chose these permissions because they are the same as [those created by Step Functions](https://docs.aws.amazon.com/step-functions/latest/dg/glue-iam.html) when a state machine with a glue task is [created in the console](https://docs.aws.amazon.com/step-functions/latest/dg/service-integration-iam-templates.html). The `BatchStopJobRun` is required for stopping the glue job if the state machine execution is cancelled.

I didn't think it was possible to specify a resource for these permissions because the Glue docs state that ["Glue doesn't allow ARNs for [...] job runs"](https://docs.aws.amazon.com/glue/latest/dg/using-identity-based-policies.html#glue-identity-based-policy-limitations). But I changed the permissions and it seems to work fine for starting the run, realizing the run has finished, and stopping the run.

Since the permissions are different for the different service integration patterns (fire and forget vs. sync), I also added a conditional that sets the appropriate set based on the specified pattern.